src/components/tooltip/Tooltip.tsx

```diff
@@ -215,7 +221,11 @@ class Tooltip extends React.Component<TooltipProps, State> {
     handleKeyDown = (event: KeyboardEvent) => {
         if (event.key === 'Escape') {
             event.stopPropagation();
-            this.setState({ isShown: false });
+            if (this.isControlled()) {
+                this.setState({ wasClosedByUser: true });
+            } else {
+                this.setState({ isShown: false });
+            }
```

Comment:
thoughts on keeping `this.setState({ isShown: false });` to always set?

even though we don't use the internal state for controlled inputs, the state is still kept in sync through the other methods: `handleMouseEnter`, `handleMouseLeave`, etc. so it might be useful in the future that it's all still in sync.

```
            this.setState({ isShown: false });
            if (this.isControlled()) {
                this.setState({ wasClosedByUser: true });
            }
```

Reply:
That sounds good